### ISSUE TITLE
Generic TypedDict fails with Generic base

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pyright#6318]
**Describe the bug**
Seems to be a mismatch between behaviour for creating Generic TypedDicts with PEP 695 syntax vs using an old-style Generic base.

**Code or Screenshots**

```python
from typing import Generic, TypeVar, TypedDict


class FooPEP695[T](TypedDict):
    boo: T

T = TypeVar("T")


class FooGenericBase(TypedDict, Generic[T]):
    boo: T


class Baz(TypedDict):
    test: int


class BarPEP695(FooPEP695[str], Baz):
    ...


class BarGenericBase(FooGenericBase[str], Baz): # fails
    ...
```

```diff
$ pyright test.py
test.py
-   test.py:22:7 - error: Cannot create consistent method ordering
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**
pyright 1.1.334

**Comment by erictraut** [comment]
Thanks for the bug report. The runtime logic that computes the MRO has some special-case logic to handle certain combinations of base classes that would otherwise conflict. Pyright already knew about some of these special cases, but it appears that a new special case was added to support generic `TypedDict` classes in Python 3.11. I've updated the special casing in pyright to reflect the runtime behavior.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.335, which I just published. It will also be included in a future release of pylance.
